test/lint/lint-git-commit-check.sh

```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env bash
+# Copyright (c) 2017-2019 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+#
+# Linter to check that commit messages have a new line before the body
+# or no body at all
+#
+# reference: gist.github.com/agnivade/67b42d664ece2d4210c7
+#
+# WARNING: Using a "+" character in commit messages can result in 
+# improper parsing
+
+export LC_ALL=C
+while read -d "+" -r msg  || [[ -n "$msg" ]]; do
+    n_line=0
+    while IFS= read -r line || [[ -n "$line" ]]; do
+        n_line=$((n_line+1))
+        length=${#line}
+        if [ $n_line -eq 2 ] && [ $length -ne 0 ]; then
+            echo "Your subject line follows a non-empty line. Subject lines should always be followed by a blank line."
```

Comment:
This message isn't quite right.  Maybe something like one of these is better.

```suggestion
            echo "Your body follows a non-empty line. Subject lines should always be followed by a blank line."
```

```suggestion
            echo "Your subject line is followed by a non-empty line. Subject lines should always be followed by a blank line."
```

Reply:
fixed